### PR TITLE
Vfat fixes

### PR DESCRIPTION
**Pull request by gurcei**

### CODE REVIEW

**Review by gardners — COMMENTED** [review]
Oddly it is only showing me the changes to freezer.c, not to the HYPPO code, although it says "6 files changed".  Have you changed the pull request after it was created?

**Comment by gurcei** [comment]
> Oddly it is only showing me the changes to freezer.c, not to the HYPPO code, although it says "6 files changed". Have you changed the pull request after it was created?

Oops, sorry didn't see your message earlier. Oh, I can see plenty of files on my side. Did you look at the "Files Changed" tab?

File: src/hyppo/dos.asm [hunk 1]
```diff
@@ -2349,6 +2351,11 @@ dff_have_next_entry:
 
 dos_opendir:
 
+        ;; assure we are using the sdcard buffer (not the fdc buffer)
+        lda $d689
```
Comment (gardners) [inline src/hyppo/dos.asm:2355]: Save 3 bytes with:

LDA #$80
TSB $D689

File: src/hyppo/dos.asm [hunk 2]
```diff
@@ -2725,17 +2749,19 @@ drce_eot_in_filename:
 
         ;; got all characters from this LFN piece
         ;;
-        cpx dos_dirent_longfilename_length
-        bcc drce_piece_didnt_grow_name_length
-        stx dos_dirent_longfilename_length
-        cpx #$3f
-        bcs drce_eot_in_filename2
-
+        lda dos_first_vfat_chunk_in_list_flag
```
Comment (gardners) [inline src/hyppo/dos.asm:2752]: Please verify this against a filename that is >64 bytes long, to make sure it doesn't do naughty things, like overwrite memory in HYPPO.  The HYPPO simulation environment thingy I made might be a good tool for automatic testing of this and other things.

**Review by gardners — CHANGES_REQUESTED** [review]
Hello Gurce,
Great work on fixing the LFN stuff. 
If you have put the toupper stuff in place such that the old upper case conversion stuff is moot, please purge that out of the source to reduce confusion.  
Also please make the change to use TSB to save a few bytes, and confirm that over-long LFNs can't cause memory corruption in HYPPO, then we are all good!

Thanks again,
Paul.

File: src/hyppo/dos.asm (hunk 1 above)
Comment (gurcei) [reply src/hyppo/dos.asm:2355]: Okidoke, will tweak this now.

File: src/hyppo/dos.asm (hunk 2 above)
Comment (gurcei) [reply src/hyppo/dos.asm:2752]: Hmm, I'm not familiar with this HYPPO simulation environment, can you fill me in on this?

Comment (gurcei) [reply src/hyppo/dos.asm:2729]: Ok, I've debugged tonight in xemu with a filename greater than 64 chars. Inside "**dos.asm**", I confirmed that:
- only max 64 chars will be read
- For any entries larger than this, no additional chars will be read and the file will be ignored and we'll move onto the next direntry of interest.

So I'm feeling we're safe on this point.

**Pull request merged**